src/Microsoft.Health.Dicom.SqlServer/Features/Query/DicomTagSqlEntry.cs

```diff
@@ -69,11 +69,11 @@ private DicomTagSqlEntry(SqlTableType sqlTableType, Column sqlColumn, string ful
 
         public Column SqlKeyColumn { get; }
 
-        public bool IsExtendedQueryTag { get; }
+        public bool IsIndexedQueryTag { get; }
 
         public static DicomTagSqlEntry GetDicomTagSqlEntry(QueryTag queryTag)
         {
-            return queryTag.IsExtendedQueryTag ? ExtendedQueryTagVRToSqlMapping[queryTag.VR] : CoreQueryTagToSqlMapping[queryTag.Tag];
+            return queryTag.IsExtendedQueryTag || queryTag.IsWorkitemQueryTag ? ExtendedQueryTagVRToSqlMapping[queryTag.VR] : CoreQueryTagToSqlMapping[queryTag.Tag];
```

Comment:
Same feedback here. There should be no single class which knows both DicomInstance and WorkItem. Common logic should be operated on Interfaces implemented for different Resource types

Reply:
Refactored to respective classes

Reply:
Changed to be resourceType agnostic